cumulus/parachains/integration-tests/emulated/tests/assets/asset-hub-westend/src/tests/set_asset_claimer.rs

```diff
@@ -0,0 +1,257 @@
+// limitations under the License.
+
+//! Tests related to claiming assets trapped during XCM execution.
+
+use crate::imports::*;
+use emulated_integration_tests_common::{
+	accounts::{ALICE, BOB},
+	impls::AccountId32,
+};
+
+use frame_support::{assert_ok, sp_runtime::traits::Dispatchable};
+use xcm_executor::traits::ConvertLocation;
+use crate::imports::bhw_xcm_config::{LocationToAccountId, UniversalLocation};
+
+#[test]
+fn test_set_asset_claimer_within_a_chain() {
+	let (alice_account, alice_location) = account_and_location(ALICE);
+	let (bob_account, bob_location) = account_and_location(BOB);
+
+	let amount_to_send = 16_000_000_000_000;
+	let assets: Assets = (Parent, amount_to_send).into();
+
+	fund_account(&alice_account, amount_to_send * 2);
+	assert_eq!(query_balance(&alice_account, &RelayLocation::get()), amount_to_send * 2);
+
+	let test_args = TestContext {
+		sender: alice_account.clone(),
+		receiver: bob_account.clone(),
+		args: TestArgs::new_para(
+			bob_location.clone(),
+			bob_account.clone(),
+			amount_to_send,
+			assets.clone(),
+			None,
+			0,
+		),
+	};
+	let test = ParaToParaThroughAHTest::new(test_args);
+	execute_test(test.clone(), bob_location.clone(), transfer_assets);
+
+	let alice_assets_after = query_balance(&alice_account, &RelayLocation::get());
+	assert_eq!(alice_assets_after, amount_to_send);
+
+	let test_args = TestContext {
+		sender: bob_account.clone(),
+		receiver: alice_account.clone(),
+		args: TestArgs::new_para(
+			alice_location.clone(),
+			alice_account.clone(),
+			amount_to_send,
+			assets.clone(),
+			None,
+			0,
+		),
+	};
+	let test = ParaToParaThroughAHTest::new(test_args);
+	execute_test(test.clone(), bob_location.clone(), claim_assets);
+
+	let bob_assets_after = query_balance(&bob_account, &RelayLocation::get());
+	assert_eq!(bob_assets_after, amount_to_send);
+}
+
+fn account_and_location(account: &str) -> (AccountId32, Location) {
+	let account_id = PenpalA::account_id_of(account);
+	let account_clone = account_id.clone();
+	let location: Location =
+		[Junction::AccountId32 { network: Some(Westend), id: account_id.into() }].into();
+	(account_clone, location)
+}
+
+fn fund_account(account: &AccountId, amount: u128) {
+	let asset_owner = PenpalAssetOwner::get();
+	PenpalA::mint_foreign_asset(
+		<PenpalA as Chain>::RuntimeOrigin::signed(asset_owner),
+		Location::parent(),
+		account.clone(),
+		amount,
+	);
+}
+
+fn query_balance(account: &AccountId, asset_location: &Location) -> u128 {
+	PenpalA::execute_with(|| {
+		type ForeignAssets = <PenpalA as PenpalAPallet>::ForeignAssets;
+		<ForeignAssets as Inspect<_>>::balance(asset_location.clone(), account)
+	})
+}
+
+fn execute_test(
+	test: ParaToParaThroughAHTest,
+	claimer: Location,
+	xcm_fn: impl Fn(ParaToParaThroughAHTest, Location) -> <PenpalA as Chain>::RuntimeCall,
+) {
+	let call = xcm_fn(test.clone(), claimer.clone());
+	PenpalA::execute_with(|| {
+		assert!(call.dispatch(test.signed_origin).is_ok());
+	});
+}
```

Comment:
nit:

you could drop this helper function and just hardcode the inner call - the helper fn is small, takes kind of the same amount of lines of code with or without it - and it has the disadvantage that the inner `assert!` will report this same line number for all failing tests.

also, please use `assert_ok!(result)` rather than `assert!(result.is_ok)`.

Reply:
done